### PR TITLE
fix(notifications): map API status field to frontend read boolean

### LINKED ISSUE #1674: Unable to delete notifications
1. Please check that you are running latest release available at https://github.com/tphakala/birdnet-go/releases
2. Please fill in this issue template in as detail as possible

## Issue Type

- [x] Bug report
- [x] Feature request
- [ ] Question / Support

## Description

When viewing notifications whether grouped or individual, selecting the delete option doesn't delete the notification and results in an unresponsive (link on the page cannot be clicked) page, only a refresh brings back the functionality.

Testing in firefox, chrome and edge on macOS 26 (all browsers up to date as of day issue posted)

## Steps to Reproduce (for bugs)

1. Browse to notifications from the notifications icon
2. Select the delete icon

## Expected behavior

Expect that the notification is deleted whether a true notification or an alert notifications

## Actual behavior

No notifications are deleted

### Version Information (build date)

Provide the following line from your system:

nightly-20251223

### System Information

- Device:
  - [x] Raspberry Pi Zero 2W
- OS: DietPi
- OS Version: v9.20.1
- Hardware specs:
  - CPU: quad-core 64-bit ARM Cortex-A53 @1ghz
  - RAM: 512mb
  - Storage type and size: 64GB Sandisk Micro SD

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

Fixes the issue where deleting notifications doesn't work and the page becomes unresponsive (#1674).

**Root Cause:** The backend API sends notifications with a `status` field (string: `'unread'`, `'read'`, `'acknowledged'`) but the frontend expects a `read` field (boolean). This mismatch caused `notification.read` to always be `undefined`, which broke:
- Delete operations (UI wouldn't update properly after delete)
- Mark-as-read operations (state checks failed)
- Visual indicators for read/unread state

## Changes

- Added `mapApiNotification` and `mapApiNotifications` helper functions in `notifications.ts` to convert between API and frontend formats
- Updated `Notifications.svelte` to map notifications when loading from API
- Updated `NotificationBell.svelte` to map both API-loaded and SSE notifications
- Added comprehensive unit tests for the mapping functions

## Test plan

- [x] Unit tests for `mapApiNotification` function
  - Maps status `"unread"` → `read: false`
  - Maps status `"read"` → `read: true`
  - Maps status `"acknowledged"` → `read: true`
  - Preserves existing `read` boolean when already set
  - Preserves all other notification fields
- [x] All 1439 frontend tests pass
- [x] `task frontend-quality` passes
- [x] Svelte MCP validation passes (no issues)
- [ ] Manual testing: Verify delete notifications works
- [ ] Manual testing: Verify mark-as-read works
- [ ] Manual testing: Verify UI state updates correctly

Closes #1674

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved notification handling consistency across different data sources to ensure read status is properly tracked.

* **Tests**
  * Added comprehensive test suite for notification utilities, covering data transformation, deduplication, grouping, and validation logic.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: frontend/src/lib/utils/notifications.ts [hunk 1]
```diff
@@ -24,6 +24,86 @@ export interface Notification {
...
+  // This handles test mocks and already-mapped notifications
+  if (typeof notification.read === 'boolean') {
+    return notification as Notification;
+  }
+
+  // Otherwise, derive 'read' from 'status'
+  // 'unread' -> read: false, 'read'/'acknowledged' -> read: true
+  return {
+    ...notification,
+    read: notification.status !== 'unread',
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/utils/notifications.ts]: Fixed. Changed to explicit status checks: `notification.status === 'read' || notification.status === 'acknowledged'`. Added test case for undefined status edge case.

File: frontend/src/lib/utils/notifications.ts [hunk 2]
```diff
@@ -24,6 +24,86 @@ export interface Notification {
...
+  } as Notification;
```
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/utils/notifications.ts:94]: Fixed in eddb8d37. Explicit status checks now properly default undefined to read: false.

File: frontend/src/lib/utils/notifications.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tphakala) [reply frontend/src/lib/utils/notifications.ts]: Fixed in eddb8d37.
